src/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -379,6 +379,37 @@ private static string GetResourceString(ExceptionResource resource, int currentD
 
             return message;
         }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static FormatException GetFormatException(string message)
+        {
+            return new FormatException(message);
+        }
+
+        public static FormatException GetFormatException_Int32()
```

Comment:
Add a `NumericType` enum and update this to a single copy of the method with a switch on it to avoid duplication.

Something like:
`public static FormatException GetFormatException(NumericType type)`